packages/server/src/tests/api/chatConversations.spec.ts

```diff
@@ -0,0 +1,238 @@
+import { context, docIds, roles } from "@budibase/backend-core"
+import type { ChatApp, ChatConversation, User } from "@budibase/types"
+import TestConfiguration from "../utilities/TestConfiguration"
+import { prepareChatConversationForSave } from "../../api/controllers/ai/chatConversations"
+
+describe("chat conversations authorization", () => {
+  const config = new TestConfiguration()
+  let userA: User
+  let userB: User
+  let chatApp: ChatApp
+  let otherChatApp: ChatApp
+  let convoA: ChatConversation
+  let convoB: ChatConversation
+  let otherAppConvo: ChatConversation
+
+  beforeAll(async () => {
+    await config.init("chat-conversation-scope")
+    userA = config.getUser()
+    userB = await config.createUser({
+      roles: {
+        [config.getProdWorkspaceId()]: roles.BUILTIN_ROLE_IDS.BASIC,
+      },
+      builder: { global: true },
+      admin: { global: false },
+    })
+
+    await context.doInWorkspaceContext(
+      config.getProdWorkspaceId(),
+      async () => {
+        const db = context.getWorkspaceDB()
+        const now = new Date().toISOString()
+        chatApp = {
+          _id: docIds.generateChatAppID(),
+          agentId: "agent-1",
+          createdAt: now,
+          updatedAt: now,
```

Comment:
I am quite sure we can't set the `updatedAt` like this, as the db implementation will override it. If these are really important, we need to use `timekeeper` for the mocks

Reply:
Or jest fakers, as you did below